contracts/zero-ex/contracts/src/transformers/PositiveSlippageFeeTransformer.sol

```diff
@@ -0,0 +1,71 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "../errors/LibTransformERC20RichErrors.sol";
+import "./Transformer.sol";
+import "./LibERC20Transformer.sol";
+
+
+/// @dev A transformer that transfers tokens to arbitrary addresses.
+contract PositiveSlippageFeeTransformer is
+    Transformer
+{
+    using LibRichErrorsV06 for bytes;
+    using LibSafeMathV06 for uint256;
+    using LibERC20Transformer for IERC20TokenV06;
+
+    /// @dev Information for a single fee.
+    struct TokenFee {
+        // The token to transfer to `recipient`.
+        IERC20TokenV06 token;
+        // Amount of each `token` to transfer to `recipient`.
+        uint256 bestCaseAmount;
+        // Recipient of `token`.
+        address payable recipient;
+    }
+
+    /// @dev Transfers tokens to recipients.
+    /// @param context Context information.
+    /// @return success The success bytes (`LibERC20Transformer.TRANSFORMER_SUCCESS`).
+    function transform(TransformContext calldata context)
+        external
+        override
+    returns (bytes4 success)
+    {
+        TokenFee[] memory fees = abi.decode(context.data, (TokenFee[]));
+
+        // Transfer tokens to recipients.
+        for (uint256 i = 0; i < fees.length; ++i) {
+            uint256 bestCaseAmount = fees[i].bestCaseAmount;
```

Comment:
How exactly do multiple positive slippage fees work?

Reply:
good question, I kept the same structure as for the `AffiliateFeeTransformer`, it can take an array of fees, but its not currently used as such. Is there a specific reason why would affiliateFee would take an array? If specific to affiliateFee we can remove it for PositiveSlippageFee. 

https://github.com/0xProject/protocol/blob/72a74e7c66e27da02dd9f4ce604ad057c740c304/contracts/zero-ex/contracts/src/transformers/AffiliateFeeTransformer.sol#L65

Reply:
IIRC we specifically got asked for multiple affiliate fees by an integrator. But we don't expose it so no one is using it and no one complained so it's all pointless.

Reply:
okay then I'm not gonna use it. Would removing it from `AffiliateFeeTransformer` reduce our gas footprint? Adding a fee already cost a lot in gas, wondering if we could optimize a bit.

Reply:
okay I've removed the `fees` array